File: sdks/js/test/diagnostics.test.ts
const { test } = require('node:test');
const assert = require('node:assert');
const { diagnoseReasoningExhaustion } = require('../dist/diagnostics');

test('diagnoseReasoningExhaustion detects reasoning exhaustion when finish_reason is length and content is empty', () => {
  const report = diagnoseReasoningExhaustion('length', 1000, 1000, '');
  assert.strictEqual(report.exhausted, true);
  assert.strictEqual(report.reasoningTokens, 1000);
  assert.ok(report.message && report.message.includes('Reasoning consumed the entire token budget'));
});

test('diagnoseReasoningExhaustion reports not exhausted when completion text was produced', () => {
  const report = diagnoseReasoningExhaustion('length', 1000, 200, 'Here is the answer');
  assert.strictEqual(report.exhausted, false);
});

test('diagnoseReasoningExhaustion reports not exhausted on normal stop', () => {
  const report = diagnoseReasoningExhaustion('stop', 50, 10, 'Hello');
  assert.strictEqual(report.exhausted, false);
});

File: sdks/go/diagnostics_test.go
package nrouter

import (
	"testing"
)

func TestDiagnoseReasoningExhaustion(t *testing.T) {
	// Exhausted case
	report := DiagnoseReasoningExhaustion("length", 1000, 1000, "")
	if !report.Exhausted {
		t.Fatalf("expected exhaustion detected")
	}
	if report.ReasoningTokens != 1000 {
		t.Errorf("expected 1000 reasoning tokens, got %d", report.ReasoningTokens)
	}

	// Normal case with content
	normal := DiagnoseReasoningExhaustion("stop", 50, 10, "Here is the answer")
	if normal.Exhausted {
		t.Fatalf("did not expect exhaustion when content is present")
	}

	// Length finish but content generated
	withContent := DiagnoseReasoningExhaustion("length", 1000, 200, "Partial answer")
	if withContent.Exhausted {
		t.Fatalf("did not expect exhaustion when text content was produced")
	}
}
